### PR TITLE
[fix][broker] Set ServiceUnitStateChannel topic compaction threshold explicitly, improve getOwnerAsync, and fix other bugs

### PR DESCRIPTION
**Pull request by heesung-sn**
### Motivation

We better set the compaction threshold of the serviceUnitStateChannel topic instead of relying on other components' init.

Also, we need to fix the issue that lookup and unload often timeout with  500 error.
### Modifications

- Set the compaction threshold of the serviceUnitStateChannel topic in the monitor thread
- Release metadataAndPayload buffer in RawBatchMessageContainerImpl.
- Automatically restart load data store producer and table view if inactive.
- Updated getOwnerAsync for better synchronization of ownership checks, deferred lookups and active broker checks
- added minor retry logic in the ExtensibleLoadManager tests

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/heesung-sn/pulsar/pull/61

### CODE REVIEW

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewImpl.java [hunk 1]
```diff
@@ -304,8 +304,14 @@ private void readTailMessages(Reader<T> reader) {
                         log.error("Reader {} was closed while reading tail messages.",
                                 reader.getTopic(), ex);
                     } else {
+                        // Retrying on the other exceptions such as NotConnectedException
```
Comment (Demogorgon314) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewImpl.java:307]: Should we push a separate PR and add a unit test for this change?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/store/TableViewLoadDataStoreImpl.java [hunk 2]
```diff
@@ -150,9 +156,34 @@ public synchronized void init() throws IOException {
...
+    }
+
+    private void validateTableView() {
+        if (tableView == null || System.currentTimeMillis() - tableViewLastUpdateTimestamp
+                > ((long) conf.getLoadBalancerReportUpdateMaxIntervalMinutes()) * 60 * 1000 * 2) {
+            tableViewLastUpdateTimestamp = 0;
+            try {
+                closeTableView();
+                startTableView();
+                log.info("Restarted tableview on {}", topic);
```
Comment (Demogorgon314) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/store/TableViewLoadDataStoreImpl.java:182]: Should we add more info about why we restart?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java [hunk 3]
```diff
@@ -894,14 +920,20 @@ private List<Metrics> getIgnoredCommandMetrics(String advertisedBrokerAddress) {
...
         try {
             initWaiter.await();
 
             // Monitor role
             // Periodically check the role in case ZK watcher fails.
             var isChannelOwner = serviceUnitStateChannel.isChannelOwner();
             if (isChannelOwner) {
+                // System topic config might fail due to the race condition
+                // with topic policy init(Topic policies cache have not init).
+                if (!configuredSystemTopics) {
```
Comment (Demogorgon314) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java:934]: Why this change can fix this issue?

Comment (Demogorgon314) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java:934]: Can you enable the topic-level policies on the integration test(`ExtensibleLoadManagerTest`) in this PR? I see it has been disabled by #21231

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/store/TableViewLoadDataStoreImpl.java [hunk 4]
```diff
@@ -150,9 +156,34 @@ public synchronized void init() throws IOException {
...
+            } catch (Exception e) {
+                log.error("Failed to restart producer on {}", topic, e);
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private void validateTableView() {
+        if (tableView == null || System.currentTimeMillis() - tableViewLastUpdateTimestamp
+                > ((long) conf.getLoadBalancerReportUpdateMaxIntervalMinutes()) * 60 * 1000 * 2) {
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/store/TableViewLoadDataStoreImpl.java:177]: we can define a static field

Comment (heesung-sn) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/store/TableViewLoadDataStoreImpl.java:177]: done.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/store/TableViewLoadDataStoreImpl.java (hunk 2 above)
Comment (heesung-sn) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/store/TableViewLoadDataStoreImpl.java:182]: updated.

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewImpl.java (hunk 1 above)
Comment (heesung-sn) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewImpl.java:307]: I don't think this is a significant logic change because we are adding a minor delay before retrying(this retrying is a very edge case). If this is a concern, I can make this a separate PR.

**Review by Demogorgon314 — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/client/impl/RawBatchMessageContainerImpl.java [hunk 5]
```diff
@@ -187,6 +187,7 @@ public ByteBuf toByteBuf() {
         idData.writeTo(buf);
         buf.writeInt(metadataAndPayload.readableBytes());
         buf.writeBytes(metadataAndPayload);
+        metadataAndPayload.release();
```
Comment (lhotari) [inline pulsar-broker/src/main/java/org/apache/pulsar/client/impl/RawBatchMessageContainerImpl.java:190]: @heesung-sn how did you find this issue? I think this could deserve a separate PR for maintenance branches? Please report a separate issue in apache/pulsar issues to track this.

**Comment by lhotari** [comment]
@heesung-sn  Is there a need to backport the ExtensibleLoadManager related fixes all the way to 3.0.x ?

File: pulsar-broker/src/main/java/org/apache/pulsar/client/impl/RawBatchMessageContainerImpl.java (hunk 5 above)
Comment (heesung-sn) [reply pulsar-broker/src/main/java/org/apache/pulsar/client/impl/RawBatchMessageContainerImpl.java:190]: Yes, actually, I am trying to see if I can cherry-pick this PR to the branch-3.0 and others.

If not easy, I will raise separate PRs to them.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImplTest.java [hunk 6]
```diff
@@ -258,9 +257,6 @@ public void testAssign() throws Exception {
         Optional<BrokerLookupData> brokerLookupData1 = secondaryLoadManager.assign(Optional.empty(), bundle).get();
         assertEquals(brokerLookupData, brokerLookupData1);
 
-        verify(primaryLoadManager, times(1)).getBrokerSelectionStrategy();
-        verify(secondaryLoadManager, times(0)).getBrokerSelectionStrategy();
-
```
[the remark this answers is not in the document]
Comment (heesung-sn) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImplTest.java:263]: the selection cannot be fully controlled because the TopicPolicy topics can be also impacted during the test.

**Comment by heesung-sn** [comment]
> @heesung-sn Is there a need to backport the ExtensibleLoadManager related fixes all the way to 3.0.x ?


Yes, I am trying to backport other PRs.

This one, too.
https://github.com/apache/pulsar/pull/22112

**Comment by heesung-sn** [comment]
@lhotari plz review this cherry-pick PR : https://github.com/apache/pulsar/pull/22154
